packages/expo/android/src/main/java/expo/modules/ApplicationLifecycleDispatcher.kt

```diff
@@ -2,19 +2,22 @@ package expo.modules
 
 import android.app.Application
 import android.content.res.Configuration
+import expo.modules.core.ModulePriorities
 
 class ApplicationLifecycleDispatcher private constructor() {
   companion object {
     @JvmStatic
     fun onApplicationCreate(application: Application) {
       ExpoModulesPackage.packageList
+        .sortedByDescending { ModulePriorities.get(it::class.qualifiedName) }
         .flatMap { it.createApplicationLifecycleListeners(application) }
         .forEach { it.onCreate(application) }
     }
 
     @JvmStatic
     fun onConfigurationChanged(application: Application, newConfig: Configuration) {
       ExpoModulesPackage.packageList
+        .sortedByDescending { ModulePriorities.get(it::class.qualifiedName) }
         .flatMap { it.createApplicationLifecycleListeners(application) }
```

Comment:
```suggestion
      getApplicationLifecycleListeners()
```

Reply:
- cache the calls to `.sortedByDescending` by kotlin lazy property: https://github.com/expo/expo/pull/14883/commits/56ae93f391c84cd6d291642b85ab647cbeb2a359

- for the `getApplicationLifecycleListeners()`, it's not that clear because `createApplicationLifecycleListeners()` needs to pass an Application and i don't want to cache the Application. anyway this is my attempt, please let me know how do you think about it: https://github.com/expo/expo/pull/14883/commits/75902ba8cec55a290794a148b2b065e4560fcd72